### ISSUE TITLE
Mangling computed properties from symbols when it should not

### ISSUE THREAD

**Issue by PaulBlanche** [fabiosantoscode/terser#93]
When mangling properties, computed properties from symbols should not be mangled.

**Minimal exemple**
running ``terser --mangle-props test.js`` with
```javascript
const mySymbol = Symbol('foo');
const myObject = {
    [mySymbol]: 'bar'
}
console.log(myObject[mySymbol]);
```
gives
```javascript
const mySymbol=Symbol("foo");
const myObject={o:"bar"};
console.log(myObject[mySymbol]);
```
instead of
```javascript
const mySymbol=Symbol("foo");
const myObject={[mySymbol]:"bar"};
console.log(myObject[mySymbol]);
```

**Comment by kzc** [comment]
Thanks for the concise report.

This `--mangle-props` bug is unrelated to symbols and would occur with any computed property. It is clearly evident when mangle properties debug is enabled:

```
$ terser -V
terser 3.8.1

$ echo '({ foo: 1, [bar]: 2, [3+4]: 7 })' | terser --mangle-props debug 
({_$foo$_:1,"_$[object Object]$_":2,"_$[object Object]$_":7});
```

Given the nature of the bug, it's surprising that it wasn't reported earlier.

Likely fix:

```patch
diff --git a/lib/propmangle.js b/lib/propmangle.js
index d9dbb8c..1aba588 100644
--- a/lib/propmangle.js
+++ b/lib/propmangle.js
@@ -158,7 +158,9 @@ function mangle_properties(ast, options) {
     // step 1: find candidates to mangle
     ast.walk(new TreeWalker(function(node) {
         if (node instanceof AST_ObjectKeyVal) {
-            add(node.key);
+            if (!(node.key instanceof AST_Node)) {
+                add(node.key);
+            }
         }
         else if (node instanceof AST_ObjectProperty) {
             // setter or getter, since KeyVal is handled above
@@ -179,7 +181,9 @@ function mangle_properties(ast, options) {
     // step 2: transform the tree, renaming properties
     return ast.transform(new TreeTransformer(function(node) {
         if (node instanceof AST_ObjectKeyVal) {
-            node.key = mangle(node.key);
+            if (!(node.key instanceof AST_Node)) {
+                node.key = mangle(node.key);
+            }
         }
         else if (node instanceof AST_ObjectProperty) {
             // setter or getter
diff --git a/test/compress/properties.js b/test/compress/properties.js
index 801e4c3..6bff08f 100644
--- a/test/compress/properties.js
+++ b/test/compress/properties.js
@@ -2336,3 +2336,35 @@ issue_t64: {
     expect_stdout: "PASS"
     node_version: ">=6"
 }
+
+dont_mangle_computed_property: {
+    mangle = {
+        properties: {}
+    }
+    input: {
+        "AAAAAAAAAAAAAAAAAA";
+        "BBBBBBBBBBBBBBBBB";
+        const prop = Symbol("foo");
+        const obj = {
+            [prop]: "bar",
+            "baz": 1,
+            qux: 2,
+            [3 + 4]: "seven",
+        }
+        console.log(obj[prop], obj["baz"], obj.qux, obj[7]);
+    }
+    expect: {
+        "AAAAAAAAAAAAAAAAAA";
+        "BBBBBBBBBBBBBBBBB";
+        const prop = Symbol("foo");
+        const obj = {
+            [prop]: "bar",
+            A: 1,
+            B: 2,
+            [3 + 4]: "seven",
+        };
+        console.log(obj[prop], obj["A"], obj.B, obj[7]);
+    }
+    expect_stdout: "bar 1 2 seven"
+    node_version: ">=6"
+}
```

@PaulBlanche Please submit a PR with the fix above.

**Comment by kzc** [comment]
The patch above does not handle numeric properties correctly. Fixed in #110.
